controllers/openstackcluster_controller.go

```diff
@@ -126,20 +126,22 @@ func (r *OpenStackClusterReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		if openStackCluster.Status.Bastion == nil {
 			openStackCluster.Status.Bastion = &infrav1.BastionStatus{}
 		}
-		changed, err := compute.ResolveReferencedMachineResources(scope, openStackCluster, &openStackCluster.Spec.Bastion.Instance, &openStackCluster.Status.Bastion.ReferencedResources)
-		if err != nil {
-			return reconcile.Result{}, err
-		}
-		if changed {
+		if openStackCluster.Status.Bastion.ReferencedResources == nil {
+			openStackCluster.Status.Bastion.ReferencedResources = &infrav1.ReferencedMachineResources{}
+			err := compute.ResolveReferencedMachineResources(scope, openStackCluster, &openStackCluster.Spec.Bastion.Instance, openStackCluster.Status.Bastion.ReferencedResources)
```

Comment:
We must not do this because it will write an empty referenced machine resources to the cluster status if resolve fails. This needs to be:

```golang
  referencedResources, err := compute.ResolveReferencedMachineResources(...)
  if err != nil {
    ...
  }
  openStackCluster.Status.Bastion.ReferencedResources = referencedResources
```